src/main/java/org/folio/innreach/domain/service/impl/InventoryStorageServiceImpl.java

```diff
@@ -0,0 +1,76 @@
+package org.folio.innreach.domain.service.impl;
+
+import static org.folio.innreach.util.ListUtils.getFirstItem;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import org.folio.innreach.client.HoldingsStorageClient;
+import org.folio.innreach.client.HridSettingsClient;
+import org.folio.innreach.client.InstanceContributorTypeClient;
+import org.folio.innreach.client.InstanceStorageClient;
+import org.folio.innreach.client.InstanceTypeClient;
+import org.folio.innreach.client.ItemsStorageClient;
+import org.folio.innreach.client.ServicePointsClient;
+import org.folio.innreach.domain.service.InventoryStorageService;
+import org.folio.innreach.dto.Holding;
+import org.folio.innreach.dto.Instance;
+import org.folio.innreach.dto.Item;
+
+@Service
+@RequiredArgsConstructor
+public class InventoryStorageServiceImpl implements InventoryStorageService {
+
+  private final InstanceStorageClient instanceStorageClient;
+  private final ItemsStorageClient itemsStorageClient;
+  private final HoldingsStorageClient holdingsStorageClient;
+  private final ServicePointsClient servicePointsClient;
+  private final HridSettingsClient hridSettingsClient;
+  private final InstanceTypeClient instanceTypeClient;
+  private final InstanceContributorTypeClient nameTypeClient;
+
+  @Override
+  public Instance queryInstanceByHrid(String instanceHrid) {
+    return getFirstItem(instanceStorageClient.queryInstanceByHrid(instanceHrid))
+      .orElseThrow(() -> new IllegalArgumentException("No instance found by hrid " + instanceHrid));
+  }
+
+  @Override
+  public ServicePointsClient.ServicePoint queryServicePointByCode(String locationCode) {
+    return getFirstItem(servicePointsClient.queryServicePointByCode(locationCode))
+      .orElseThrow(() -> new IllegalStateException("Service point is not found for pickup location code: " + locationCode));
```

Comment:
In the previous method you use IllegalArgumentException to notify something is not found, here it is IllegalStateException -- doesn't look consistent